spec/models/product_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+RSpec.describe Product, type: :model do
+  let!(:product){ FactoryBot.create(:product) }
+  let!(:product_1){ FactoryBot.create(:product) }
+  describe "associations" do
+    context "with belong to" do
+      it {is_expected.to belong_to(:category) }
+    end
+
+    context "with has many" do
+      it { is_expected.to have_many(:votes).dependent(:destroy) }
+      it { is_expected.to have_many(:product_attributes).dependent(:destroy) }
+      it { is_expected.to have_many(:product_images).dependent(:destroy) }
+    end
+  end
+
+  describe "validations" do
```

Comment:
describe, context và it có ý nghĩa gì

Reply:
- Describe dùng để nhóm các test case theo class, module, method hoặc theo action của controller.
Cần có mô tả cho trường hợp không là method.
Trong trường hợp đối với instance method thì gắn thêm "#" ví dụ "#method".
Trong trường hợp đối với class method thì gắn thêm "." ví dụ ".method".
- Context dùng để nhóm các điều kiện, thường bắt đầu bằng when hoăc with
- It tương ứng với 1 test case.